### ISSUE TITLE
When building on Windows w/swiftc, can accidently include both msvc and gcc math.inls.

### ISSUE THREAD

**Issue by jeffdav** [awslabs/aws-c-common#1155]
### Describe the bug

On Windows, I'm building `aws-sdk-swift` using CMake.  I build `aws-c-*` using `ExternalProject_Add()`. 
(Thank you for providing `CMakeLists.txt` for your projects!)

The issue is when building `aws-sdk-swift`, we end up using clang, which causes `math.h` include both `math.msvc.inl` and `math.gcc_builtin.inl` during module import, leading to duplicate symbols.

### Expected Behavior

We should only include one or the other. 

### Current Behavior

We include both.

### Reproduction Steps

Build `aws-sdk-swift` using CMake on Windows.  I'm happy to share my `CMakeLists.txt` for doing this if it is helpful.  

### Possible Solution

Currently I'm patching:

```c
#if defined(__clang__) || defined(__GNUC__)
```

To:
```c
#if (defined(__clang__) || defined(__GNUC__)) && !defined(AWS_HAVE_MSVC_INTRINSICS_X64)
#    include <aws/common/math.gcc_builtin.inl>
#endif
```

I'm not 100% certain this is the most correct, but it works.

### aws-c-common version used

commit 69789467ebb76526bffc8ad0e552858256b6a326 (origin/main, origin/HEAD, main)

### Compiler and version used

swiftc: compnerd.org Swift version 6.0-dev (LLVM a527f49f462b0d7, Swift eef85a7be270421), clang: compnerd.org clang version 17.0.6, cl: Microsoft (R) C/C++ Optimizing Compiler Version 19.41.34120 for x64

### Operating System and version

Windows 11 23H2

**Comment by TingDaoK** [comment]
Yeah, please share the CMakeLists, which will help us to reproduce the issue.

**Comment by jmklix** [comment]
Please note that it is still a feature request to get swiftc working on windows. As mentioned [here](https://github.com/awslabs/aws-crt-swift/issues/289#issuecomment-2391869469) by @waahm7
> Currently, both Aws-crt-swift and Aws-sdk-swift don't support Windows. This is a feature request, and I think we will probably hit more issues than just https://github.com/awslabs/aws-crt-swift/pull/290 to properly support Windows and also add CI for it. Swift SDK might have to do some work as well on their side. We have added this to our backlog and will update here as soon as we have any updates.
>
> Please 👍 the original issue and that will help us prioritize this.

**Comment by jjerphan** [comment]
I also have this issue when building using `clang-cl` a project depending transitively on `aws-c-common` (see https://github.com/man-group/ArcticDB/commit/9323534f077a0228cc255f2af91657d2297cd3b8 from https://github.com/man-group/ArcticDB/pull/2252).

Might `AWS_HAVE_MSVC_INTRINSICS_X64` and `__clang__` be defined by `clang-cl`?

**Comment by jjerphan** [comment]
https://github.com/awslabs/aws-c-common/pull/1194 was opened to propose a solution.

**Comment by graebm** [comment]
Does https://github.com/awslabs/aws-c-common/pull/1195 fix the double definition?

No promises there aren't more problems for Swift, because Swift can't use CMake, and aws-c-common requires CMake, as mentioned above, to detect various features.

For the ArcticDB comment, looks like it's [pulling in aws-c-common via vcpkg](https://github.com/man-group/ArcticDB/blob/4f3ce004cbed8613b7600b7ffecc5152b51c8388/cpp/vcpkg.json#L57)? I don't know anything about vcpkg, but based on a quick Google, it's CMake-aware, so *should* work. Does vcpkg pull down precompiled binaries? Are those precompiled binaries compiled with MSVC? If so, and you're building your project with clang-cl, I'm guessing you're going to have a bad time. I would build everything with the same compiler.

**Comment by jjerphan** [comment]
> Does https://github.com/awslabs/aws-c-common/pull/1195 fix the double definition?

I guess, it might?

As for ArcticDB, it is built with static linkage using vcpkg to get dependencies for its PyPI wheels, but built with shared linkage using conda-forge to get dependencies to be distributed there. I was not using `cl` but `clang-cl` as a workaround temporarily.

**Comment by graebm** [comment]
I merged the PR and made a new release [v0.12.2](https://github.com/awslabs/aws-c-common/releases/tag/v0.12.2). Let me know if that fixes it
